src/task-4.js

```diff
@@ -1,9 +1,56 @@
 
 export default function runRover(commandsList) {
-  // Change me!
+  const ship = {
+    directions: ['top', 'left', 'bottom', 'right'],
+    indexOfCurrDirection: 0,
+    x: 0,
+    y: 0,
+    go(distance) {
+      switch (this.directions[this.indexOfCurrDirection]) {
+        case 'top': this.y += distance;
+          break;
+        case 'left': this.x -= distance;
+          break;
+        case 'bottom': this.y -= distance;
+          break;
+        case 'right': this.x += distance;
+          break;
+        default: break;
+      }
+    },
+    turn(direction) {
+      if (direction === 'left') {
+        this.indexOfCurrDirection++;
+        if (this.indexOfCurrDirection === 4) { this.indexOfCurrDirection = 0; }
+      } else {
+        this.indexOfCurrDirection--;
+        if (this.indexOfCurrDirection === -1) { this.indexOfCurrDirection = 3; }
+      }
+    }
+  };
+  const commandsInCorrectOrder = [];
+  let minValue = Infinity;
+  commandsList.forEach(currentCommand => {
+    if (currentCommand.order < minValue) {
+      minValue = currentCommand.order;
+    }
+  });
+  commandsList.forEach(currentCommand => {
+    commandsInCorrectOrder[currentCommand.order - minValue] = currentCommand.command;
+  });
+  console.log(commandsInCorrectOrder);
 
+  commandsInCorrectOrder.forEach(currentCommand => {
```

Comment:
With the current implementation, it will be hard to get ride of `forEach` here, I suggest to make `go` and `turn` methods 
one f-n functions with will accept start position and direction and then you can call `reduce` on `commandsInCorrectOrder` with this new function